src/gui/socketapi/socketapi.cpp

```diff
@@ -479,7 +476,9 @@ void SocketApi::slotUpdateFolderView(Folder *f)
 
 void SocketApi::broadcastMessage(const QString &msg, bool doWait)
 {
-    for (const auto &listener : qAsConst(_listeners)) {
+    // operate on a copy of the list to prevent concurrency issues
+    const auto clients = _listeners.values();
```

Comment:
The Qt docs don't suggest that this method is thread-safe. I think there is no way to get around using a mutex.